lib/dependencies/ImportMetaDependency.js

```diff
@@ -0,0 +1,119 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Ivan Kopeykin @vankop
+*/
+
+"use strict";
+
+const NullDependency = require("./NullDependency");
+
+/** @typedef {import("webpack-sources").ReplaceSource} ReplaceSource */
+/** @typedef {import("../Dependency")} Dependency */
+/** @typedef {import("../DependencyTemplate").DependencyTemplateContext} DependencyTemplateContext */
+/** @typedef {import("../Module")} Module */
+/** @typedef {import("../NormalModule")} NormalModule */
+
+/** @enum {number} */
+const ImportMetaProperty = {
+	Url: 0
+};
+
+class ImportMetaDependency extends NullDependency {
+	/**
+	 * @param {[number, number]} range range
+	 * @param {ImportMetaProperty=} property import.meta property
+	 */
+	constructor(range, property) {
+		super();
+		this.range = range;
+		this.property = property;
+	}
+
+	get type() {
+		return "import.meta";
+	}
+
+	get category() {
+		return "esm";
+	}
+
+	serialize({ write }) {
+		write(this.range);
+		write(this.property);
+	}
+
+	deserialize({ read }) {
+		this.range = read();
+		this.property = read();
+	}
+}
+
+// makeSerializable(
```

Comment:
getting weird behavior with uncommented code (need to import `makeSerializable` from `../util/makeSerializable`)

```
const makeSerializable = require("../util/makeSerializable");
// makeSerializable === {} 
// So it is not a function..
```
Maybe I doing something wrong...